src/SFA.DAS.PAS.Account.Api/Orchestrator/EmailOrchestrator.cs

```diff
@@ -15,65 +17,111 @@ public class EmailOrchestrator
         private readonly IAccountOrchestrator _accountOrchestrator;
         private readonly IMediator _mediator;
         private readonly IIdamsEmailServiceWrapper _idamsEmailServiceWrapper;
-        private readonly ProviderApprenticeshipsServiceConfiguration _configuration;
+        private readonly IProviderCommitmentsLogger _logger;
 
-        public EmailOrchestrator(IAccountOrchestrator accountOrchestrator, IMediator mediator, IIdamsEmailServiceWrapper idamsEmailServiceWrapper, ProviderApprenticeshipsServiceConfiguration configuration)
+        public EmailOrchestrator(IAccountOrchestrator accountOrchestrator, IMediator mediator, IIdamsEmailServiceWrapper idamsEmailServiceWrapper, IProviderCommitmentsLogger logger)
         {
             _accountOrchestrator = accountOrchestrator;
             _mediator = mediator;
             _idamsEmailServiceWrapper = idamsEmailServiceWrapper;
-            _configuration = configuration;
+            _logger = logger;
         }
 
-        public async Task SendEmailToAllProviderRecipients(long ukprn, ProviderEmailRequest message)
+        public async Task SendEmailToAllProviderRecipients(long providerId, ProviderEmailRequest message)
         {
-            List<string> recipients = new List<string>();
+            var recipients = new List<string>();
 
-            if (!_configuration.CommitmentNotification.UseProviderEmail)
+            Task<List<string>> idamsUsersTask;
+            Task<List<string>> idamsSuperUsersTask;
+
+            var idamsError = false;
+            try
+            {
+                _logger.Info($"Retrieving DAS Users and Super Users from Provider IDAMS for Provider {providerId}");
+                idamsUsersTask = _idamsEmailServiceWrapper.GetEmailsAsync(providerId);
+                idamsSuperUsersTask = _idamsEmailServiceWrapper.GetSuperUserEmailsAsync(providerId);
+
+                await Task.WhenAll(idamsUsersTask, idamsSuperUsersTask);
+            }
+            catch (Exception ex)
             {
-                recipients = _configuration.CommitmentNotification.ProviderTestEmails;
+                idamsError = true;
+                _logger.Error(ex, "An error occurred retrieving users from Provider IDAMS");
+                idamsUsersTask = Task.FromResult(new List<string>());
+                idamsSuperUsersTask = Task.FromResult(new List<string>());
             }
 
+            var idamsUsers = await idamsUsersTask;
+            var idamsSuperUsers = await idamsSuperUsersTask;
+            var allIdamsUsers = idamsUsers.Concat(idamsSuperUsers).Distinct().ToList();
+            _logger.Info($"{allIdamsUsers.Count} total users retrieved from IDAMS for Provider {providerId} ({idamsUsers.Count} DAS Users; {idamsSuperUsers.Count} Super Users)");
+
+            var accountUsers = (await _accountOrchestrator.GetAccountUsers(providerId)).ToList();
+
+            if (!idamsError) {  await RemoveAccountUsersNotInIdams(accountUsers, allIdamsUsers); }
```

Comment:
I can see we are marking users who have been removed from Idams here. But what happens is the user is removed from Idams and then re-added? Do we undelete those users. Or will the normal options where opt out/in get re-applied?

Reply:
They get undeleted.